azure/src/main/java/org/apache/iceberg/azure/blob/AzureBlobClientFactory.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.azure.blob;
+
+import com.azure.core.credential.AzureSasCredential;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import java.util.Optional;
+import org.apache.iceberg.azure.AuthType;
+import org.apache.iceberg.azure.AzureProperties;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AzureBlobClientFactory {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AzureBlobClientFactory.class);
+
+  private AzureBlobClientFactory() {
+  }
+
+  public static BlobClient createBlobClient(AzureURI azureURI, AzureProperties azureProperties) {
+    final String storageAccount = azureURI.storageAccount();
+    final BlobClientBuilder builder = new BlobClientBuilder();
```

Comment:
In Iceberg, we don't use `final` for local variables. Recent JVM versions (8+) handle this without problems and it is also not very valuable because it isn't actually translated into bytecode.

Reply:
Ack, I'll remove it for the local variables.